### DISCUSSION TITLE
How to check if a request was from the StoreApi?

### DISCUSSION THREAD

**Discussion by multiplehats** [woocommerce/woocommerce#45011] [WooCommerce, Blocks, and Site Editing]
Hi all! 

I'm trying to filter something using the woocommerce_add_cart_item_data hook whenever I'm adding something to the cart with the Store API.

However, the filter is lacking context. Usually (the old way) I would check the $_POST for my field but that's obviously not there.

I also can't pass additional parameters to the cart/add-item endpoint.

Am I missing something?

**Comment by opr** [comment]
Hi @multiplehats thanks for your question!

you could use the [`rest_pre_dispatch`](https://developer.wordpress.org/reference/hooks/rest_pre_dispatch/) filter to check this, you'd need to check the route to ensure `/wc/store` was present.

```php
add_filter( 'rest_pre_dispatch', function( $result, $server, $request ) {
    if ( str_contains( $request->get_route(), '/wc/store/v1' ) ) {
        // This was a request to the store.
    }
    return $result;
}, 10, 3 );
```

> I also can't pass additional parameters to the cart/add-item endpoint.

Please could you clarify a little more what you're trying to do? And how did you try to do this already? It looks like this endpoint only accepts id, quantity, and variation data.

**Reply by multiplehats** [reply]
Hi @opr ! Thanks for replying. 

That filter is helpful. And one I didn't know yet.

Basically what I'm trying to do is show an upsell in a custom block (unrelated to the WC blocks). I want to pass additional cart item metadata so that I can show in the WC reports that this upsell was successfully purchased thanks to my custom block. 

But the add item endpoint doesn't take additional metadata which allows me to do so.

Unless I'm approaching this the wrong way. 

Thoughts? Thanks

**Reply by opr** [reply]
Hey @multiplehats, oops I realise I made an error in the initial code example I've updated it now (fixed the path, and use `str_contains`).

I understand you want to send some data along with the `cart/add-item` endpoint.

So, this isn't something we _officially_ support, and may be changed in future so do proceed with caution, but this just worked for me:

Request to `wc/store/v1/cart/add-item`:
```
{
	"id": 16,
        "quantity": 1,
	"sneaky_data": 1234
}
```

```php
add_filter( 'rest_pre_dispatch', function( $result, $server, $request ) {
	if ( str_contains( $request->get_route(), '/wc/store/v1/cart/add-item' ) ) {
		// This was a request to the store
		$params = $request->get_params();
		if ( isset( $params['sneaky_data'] ) ) {
			// Do something with the sneaky data.
		}
	}
	return $result;
}, 10, 3 );
```

**Reply by multiplehats** [reply]
Veryyyy clever. This is absolutely perfect. Thank you so much @opr . That's all I needed to know 👏
